src/ReactWrapper.jsx

```diff
@@ -208,7 +208,10 @@ class ReactWrapper {
       throw new Error('ReactWrapper::unmount() can only be called on the root');
     }
     this.single('unmount', () => {
-      this.component.setState({ mount: false });
+      this.component.setState({
+        mount: false,
+        lastRender: this.component._reactInternalInstance._renderedComponent.getPublicInstance(),
```

Comment:
I like that idea but it leaves the question as to where to put it. Because of what the lifecycle does to `_reactInternalInstance` and the fact that `unmount()` is on `ReactWrapper` but `getInstance()` is on `ReactWrapperComponent` it seems to me like the only easy options are to put a property on `this.component`, which could certainly be a `WeakMap`, or to put it in the components state.

Without entirely knowing how they're being used elsewhere in the library there doesn't seem to be a particularly good reason that `getInstance()` and `getWrapperComponent()` are on the `ReactWrapperComponent` at all. They aren't a part of the public api and only seem to be getting used by `ReactWrapper`. With that I suppose another option would be to factor those methods out of `ReactWrapperComponent` which would allow for storing a reference to the last render on `ReactWrapper` instead. It would also make that same reference available to `mount()` which might help address #967.

I'd be happy to give that a shot but it seems a lot more involved and might reach a bit beyond a quick bug fix.